pkg/apis/servicecatalog/types.go

```diff
@@ -775,9 +775,11 @@ type ServiceInstanceSpec struct {
 	// been made to the secrets from which the parameters are sourced.
 	UpdateRequests int64
 
-	// DeletionRequested provides an imperative means to request
-	// deletion by doing an update.
-	DeletionRequested bool
+	// DeletionRequested provides a mechanism for requesting a delete without
+	// performing a direct delete (which cannot be undone in cases where the
+	// delete is rejected by the broker). This is the recommended way to
+	// delete, to avoid resources becoming stuck in the deletion phase.
+	DeletionRequested metav1.Time
```

Comment:
I still think this should be `deprovisionTimestamp`

Reply:
with this being a timestamp, do we expect people to actually set the time? That seems odd to me. And what does its value mean? What if its Jan 1, 1970? What is its Dec 31, 2020? Are we really supporting scheduled deletes? Is that even a thing in kube?

Reply:
people should not set the time, svcat would, and if we can hijack the k8s delete command, that would.

time before now means do a delete. So if Jan 1, 1970 and the resource is not deleted, try to delete it. k8s does not support a future date, see [deletionTimestamp](https://kubernetes.io/docs/reference/federation/v1/definitions/)

Reply:
Did we agree to hijack delete?  And we need it to work with vanilla kubectl, so I do not think asking people to set a time is a good UX, when 99+% of the time I bet they mean "do it now".

Reply:
I think it is bad UX to not have a time tagged to when you attempted that action. How do you know if the status changed before of after a delete was requested? How do you know when service catalog should use a bigger hammer to unbind or deprovision the thing?

the validator could hijack the setting of these field to be set to now. so user sets it to anything they want, including true or now, and then the validator replaces that with time.Now().

Reply:
We are outside the area of calling DELETE directly on any of our objects.

Reply:
Personally I am fine with keeping the name as it, with a single field that's just a timestamp and relying on svcat to handle calling it appropriately.

Reply:
If we stay with an api server, the clear choice is to make a sub-resource for each type to do the deprovision or unbind, and those methods modify this field. then the user flow is `kubectl unbind ServiceInstance foobar` but in the direction service catalog should go using CRDs that is not an option, so I am saying the existence of the field, even an empty string, allows the admission controller to set it to a correct timestamp.